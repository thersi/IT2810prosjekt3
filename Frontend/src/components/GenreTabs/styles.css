.genreTabs {
    display: flex;
    justify-content: center;
}

