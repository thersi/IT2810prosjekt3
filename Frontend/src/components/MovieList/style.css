.movieContainer {
  display: flex;
  flex-wrap: wrap;
  flex-direction: column;
  gap: 10px;
  align-content: center;
  justify-content: center;
  padding-left: 5%;
  padding-right: 5%;
}

