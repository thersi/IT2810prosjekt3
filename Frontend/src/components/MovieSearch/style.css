.pagination {
    display: flex;
    flex-wrap: wrap;
    flex-direction: column;
    margin: 15px;
    justify-content: center;
    align-content: center;
}