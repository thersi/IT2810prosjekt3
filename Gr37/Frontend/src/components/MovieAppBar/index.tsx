import AppBar from "@mui/material/AppBar";
import Box from "@mui/material/Box";
import Toolbar from "@mui/material/Toolbar";
import IconButton from "@mui/material/IconButton";
import Typography from "@mui/material/Typography";
import MenuIcon from "@mui/icons-material/Menu";
import SearchBar from "../SearchBar";
import MovieFilter from "../MovieFilter";
import { createTheme } from "@mui/system";

export default function MovieAppBar() {
  const darkTheme = createTheme({
    palette: {
      mode: "dark",
    },
  });
  return (
    <Box sx={{ flexGrow: 1 }}>
      <AppBar position="static">
        <Toolbar>
          <Typography
            variant="h6"
            noWrap
            component="div"
            sx={{ flexGrow: 1, display: { xs: "none", sm: "block" } }}
          >
            Movies
          </Typography>
          <MovieFilter />
          <SearchBar />
        </Toolbar>
      </AppBar>
    </Box>
  );
}
