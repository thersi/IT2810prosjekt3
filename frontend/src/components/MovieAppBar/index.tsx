import AppBar from "@mui/material/AppBar";
import Box from "@mui/material/Box";
import Toolbar from "@mui/material/Toolbar";
import IconButton from "@mui/material/IconButton";
import Typography from "@mui/material/Typography";
import MenuIcon from "@mui/icons-material/Menu";
import SearchBar from "../SearchBar";
import MovieFilter from "../MovieFilter";

export default function MovieAppBar() {
  return (
    <Box sx={{ flexGrow: 1 }}>
      <AppBar position="static">
        <Toolbar>
          <Typography
            variant="h6"
            noWrap
            component="div"
            sx={{ flexGrow: 1, display: { xs: "none", sm: "block" } }}
          >
            Movies
          </Typography>
          <MovieFilter />
          <SearchBar />
        </Toolbar>
      </AppBar>
    </Box>
  );
}
