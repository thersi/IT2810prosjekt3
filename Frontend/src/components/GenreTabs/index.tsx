import React, { useState } from "react";
import InputLabel from "@mui/material/InputLabel";
import MenuItem from "@mui/material/MenuItem";
import FormControl from "@mui/material/FormControl";
import Select, { SelectChangeEvent } from "@mui/material/Select";
import "./styles.css";

const genres = [
  "ACTION",
  "ADVENTURE",
  "ANIMATION",
  "BIOGRAPHY",
  "COMEDY",
  "CRIME",
  "DOCUMENTARY",
  "DRAMA",
  "FAMILY",
  "FANTASY",
  "FILMNOIR",
  "HISTORY",
  "HORROR",
  "MUSIC",
  "MUSICAL",
  "MYSTERY",
  "ROMANCE",
  "SCIFI",
  "SHORT",
  "SPORT",
  "THRILLER",
  "WAR",
  "WESTERN",
];

export default function GenreTabs(props: any) {
  const [genre, setGenre] = React.useState("");

  const handleChange = (event: SelectChangeEvent) => {
    props.handleGenre(event.target.value);
    setGenre(event.target.value);
  };

  return (
    <div className='genreTabs'>
      <FormControl id = "filterForm" variant="filled" sx={{ m: 1, minWidth: 300 }}>
        <InputLabel id="demo-simple-select-standard-label">
          Select Genre
      </InputLabel>
        <Select
          labelId="demo-simple-select-standard-label"
          id="demo-simple-select-standard"
          value={genre}
          onChange={handleChange}
          label="genre"
          autoWidth
        >
          {genres.map((movieGenre) => (
            <MenuItem id = "genreItem" value={movieGenre}>{movieGenre}</MenuItem>
          ))}
        </Select>
      </FormControl>
    </div>
  );
}
