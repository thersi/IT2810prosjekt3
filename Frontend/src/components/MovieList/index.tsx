import { Movie, MovieListProps } from "../../Interfaces";
import SingleDisplay from "../SingleDisplay";
import QueryMoviesResult from "../MovieSearch";
import "./style.css";

const MovieList = (props: MovieListProps) => {
  /* const { loadingTitle, errorTitle, dataTitle } = useQuery(
    QUERY_MOVIE_BYTITLE,
    {
      variables: { title: search },
    }
  ); */
  const { data } = props;

  return (
    <div className="movieContainer">
      {props.data.searchAndFilter.map((movie: Movie) => (
        <SingleDisplay {...movie} />
      ))}
    </div>
  );
};

export default MovieList;
