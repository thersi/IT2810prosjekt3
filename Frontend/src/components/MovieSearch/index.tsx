import { useState } from "react";
import MovieAppBar from "../MovieAppBar";
import GenreTabs from "../GenreTabs";
import "./style.css";
import { useQuery } from "@apollo/client";
import MovieList from "../MovieList";
import Pagination from "@mui/material/Pagination";
import {
  QueryMoviesInput,
  QueryMoviesResult,
} from "../../Interfaces/Interfaces";
import { QUERY_ALL_MOVIES } from "../../Queries/queries";

const MovieSearch = () => {
  const [genreValue, setGenre] = useState<string>("");
  const [sortValue, setSort] = useState<number>(1);
  const [filterValue, setFilter] = useState<string>("");
  const [searchValue, setSearch] = useState<string>("");
  const [page, setPage] = useState<number>(1);
  const limit: number = 5;

  /* The below functions are used to determine values sent in to query*/
  const handleGenre = (value: string) => {
    setGenre(value);
    setPage(1);
  };

  const handleSort = (value: boolean) => {
    setSort(value ? 1 : -1);
  };

  const handleFilter = (value: string) => {
    setFilter(value);
    setPage(1);
  };
  const handleSearch = (value: string) => {
    setSearch(value);
    setPage(1);
  };
  /* Calls on query, getting values defined in child components */
  const { data, loading, refetch } = useQuery<
    QueryMoviesResult,
    QueryMoviesInput
  >(QUERY_ALL_MOVIES, {
    variables: {
      filterGenre: genreValue,
      limit: limit,
      page: page,
      order: sortValue,
      sortOn: filterValue,
      word: searchValue,
    },
  });

  return (
    <div>
      {/* Send in props to child components so that they can send user selected info back to parent */}
      <MovieAppBar
        aria-label={"App Bar"}
        aria-required="true"
        handleSearch={handleSearch}
        handleSort={handleSort}
        handleFilter={handleFilter}
      />
      <GenreTabs
        aria-label={
          "A markdown menu with all movie genres, used for filtering movies by genre."
        }
        aria-required="true"
        handleGenre={handleGenre}
      />
      {loading ||
      typeof data ===
        "undefined" /* If query data is being loaded or is undefined */ ? (
        <p>Loading...</p>
      ) : (
        <div>
          <MovieList data={data.searchAndFilter.movies} refetch={refetch} />
          <Pagination
            className="pagination"
            count={Math.ceil(data.searchAndFilter.pages / limit)}
            defaultPage={page}
            showFirstButton
            showLastButton
            onChange={(event, value) => {
              setPage(value);
            }}
          />
        </div>
      )}
    </div>
  );
};

export default MovieSearch;
