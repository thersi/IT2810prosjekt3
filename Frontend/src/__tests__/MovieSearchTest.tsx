import EnzymeToJson from "enzyme-to-json";
import { configure, render, shallow } from "enzyme";
import MovieAppBar from "../components/MovieAppBar";
import Adapter from "enzyme-adapter-react-16";
import MovieSearch from "../components/MovieSearch";
import MovieList from "../components/MovieList";
import GenreTabs from "../components/GenreTabs";

const component = <MovieSearch />;
configure({ adapter: new Adapter() });
describe("Movie Search", () => {
  it("renders correctly", () => {
    render(component);
  });
  it("Should match snapshot", () => {
    const snapshotCheck = shallow(component);
    expect(EnzymeToJson(snapshotCheck)).toMatchSnapshot();
  });

  it("Should contain components", () => {
    const shallowComp = shallow(component);
    expect(shallowComp.find(MovieList).exists()).toBeTruthy();
    expect(shallowComp.find(MovieAppBar).exists()).toBeTruthy();
    expect(shallowComp.find(GenreTabs).exists()).toBeTruthy();
  });
});
