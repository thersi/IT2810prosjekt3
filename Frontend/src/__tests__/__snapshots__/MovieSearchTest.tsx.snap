// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Search Should match snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
